go-runtime/ftl/ftltest/ftltest.go

```diff
@@ -5,46 +5,95 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"os"
 	"reflect"
 
 	"github.com/TBD54566975/ftl/backend/schema"
+	cf "github.com/TBD54566975/ftl/common/configuration"
 	"github.com/TBD54566975/ftl/go-runtime/ftl"
-	"github.com/TBD54566975/ftl/go-runtime/modulecontext"
 	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/modulecontext"
 )
 
-type Options struct {
+type OptionsState struct {
 	configs                 map[string][]byte
 	secrets                 map[string][]byte
+	databases               map[string]modulecontext.Database
 	mockVerbs               map[schema.RefKey]modulecontext.MockVerb
 	allowDirectVerbBehavior bool
 }
 
+type Option func(context.Context, *OptionsState) error
+
 // Context suitable for use in testing FTL verbs with provided options
-func Context(options ...func(*Options) error) context.Context {
+func Context(options ...Option) context.Context {
```

Comment:
Do we have any integration tests for these new features yet?

Reply:
In integration tests there's one called `TestModuleUnitTests` which calls the unit tests in the module `wrapped`. It tests a few cases around `WithConfig`, `WithSecret`, `WithCallsAllowedWithinModule`, `WhenVerb` and `WithProjectFile`